### PR TITLE
Add isPartOf schema to content with document collections

### PR DESCRIPTION
**Pull request by MahmudH**
- Trello card: https://trello.com/c/zl1Hk6Ne
- Details of the schema can be found from https://schema.org/isPartOf
- Indicates an item or CreativeWork that this item is part of
- The pages effected are the collection pages linked to the Article schema, these include:
1. https://www.gov.uk/government/publications/toluene-properties-incident-management-and-toxicology
2. https://www.gov.uk/government/publications/driving-in-the-eu-if-theres-no-brexit-deal
3. https://www.gov.uk/government/publications/travelling-to-the-eu-with-a-uk-passport-if-theres-no-brexit-deal
- Adding this schema should help link the pages so that searches in Google are more effective and pages linked together are easier to find through Google search.

Google Structured Data Testing tool:

screen shot 2018-09-27 at 11 30 46

Current Google search results:

screen shot 2018-09-27 at 11 49 22

---

Component guide for this PR:
https://govuk-publishing-compon-pr-542.herokuapp.com/component-guide/

### CODE REVIEW

File: lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb [hunk 1]
```diff
@@ -69,20 +70,21 @@ def publishing_organisation
...
+      def step_by_step_schemas
         # We could include `related_to_step_navs` eventually too, but initially
         # link to those that we render in the "step_by_step_nav_related" component
-        @step_by_steps ||= page.content_item.dig("links", "part_of_step_navs").to_a
+        @step_by_steps ||= fetch_step_by_steps
       end
 
-      def step_by_step_schemas
-        linked_step_by_steps.map do |step_by_step|
+      def fetch_step_by_steps
```
Comment (oscarwyatt) [inline lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb]: I'd rename this to `fetch_step_by_step_schemas` so it's more consistent with `step_by_step_schemas` which makes it slightly easier for someone to grasp that the two methods are highly related

File: lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb [hunk 2]
```diff
@@ -69,20 +70,21 @@ def publishing_organisation
       end
 
       def is_part_of
-        return {} unless linked_step_by_steps.any?
+        return {} unless step_by_step_schemas.any? || collection_pages("document_collections").any?
+
```
Comment (oscarwyatt) [inline lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb]: using `collection_pages("document_collections")` here and then `document_collections` below is a little confusing and duplicates code. If someone changed that the bits you want are somewhere other than "document_collections" then you'd have to change that in two places (here and in `document_collections`). I'd suggest DRYing it up by changing it to `document_collections.any?`

File: lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb [hunk 3]
```diff
@@ -100,18 +102,22 @@ def linked_page(step_by_step)
...
+        return {} unless collection_pages("documents").any?
         {
-            "hasPart" => collection_pages.map { |document| HasPartSchema.new(document).structured_data }
+            "hasPart" => collection_pages("documents").map { |document| HasPartSchema.new(document).structured_data }
         }
       end
 
-      def collection_pages
-        @pages ||= fetch_collection_pages
+      def document_collections
```
Comment (oscarwyatt) [inline lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb]: If, as per my suggestion for line 74, you call this method twice, you could memoize this as you've done with `step_by_step_schemas` and `fetch_step_by_step_schemas`

File: lib/govuk_publishing_components/presenters/machine_readable/is_part_of_schema.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module GovukPublishingComponents
+  module Presenters
+    class IsPartOfSchema
+      attr_reader :is_part_of_url
```
Comment (oscarwyatt) [inline lib/govuk_publishing_components/presenters/machine_readable/is_part_of_schema.rb]: This is quite pernickety on my part but this attr_reader isn't necessary (nothing else uses it) and nothing else really needs to use it so I'd remove it

File: lib/govuk_publishing_components/presenters/machine_readable/is_part_of_schema.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      def initialize(is_part_of_url)
+        @is_part_of_url = is_part_of_url
+      end
+
+      def structured_data
+        # http://schema.org/isPartOf - minimal
+        {
+            "@context" => "http://schema.org",
+            "@type" => "CreativeWork",
+            "sameAs" => is_part_of_url
```
Comment (oscarwyatt) [inline lib/govuk_publishing_components/presenters/machine_readable/is_part_of_schema.rb]: If doing above, this needs to become `"sameAs" => @is_part_of_url`

File: lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb (hunk 2 above)
Comment (MahmudH) [reply lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb]: 👍

File: lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb (hunk 3 above)
Comment (MahmudH) [reply lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb]: 👍

**Pull request merged**